### DISCUSSION TITLE
Ranger

### DISCUSSION THREAD

**Discussion by mimame** [catppuccin/catppuccin#1849] [Port Requests]
https://ranger.github.io/
**Are you available to help in the creation of this port? 😄 (optional)**
Maybe (I can help but I'm a bit colorblind)

**Additional context 📷 (optional)**
Add any other context or screenshots about the port request here.

**Comment by Paulo-Murilo-Dev** [comment]
![seu filho da puta já é](https://github.com/catppuccin/catppuccin/assets/116845901/7193678f-a749-4282-a114-cebb16b0d1c0)
LETSGOO!!!

**Comment by dfrico** [comment]
I was about to open a port request but I found this one first.

I guess Ranger offers [documentation](https://github.com/ranger/ranger/blob/master/doc/colorschemes.md) and [an example](https://github.com/ranger/colorschemes/blob/master/zenburn.py) for color schemes. There's also community colorschemes such as [Dracula](https://github.com/dracula/ranger/blob/master/dracula.py) that can be used as a reference.

**Comment by lrnzdc3696** [comment]
do it

**Comment by computergnome99** [comment]
Hey, any updates on this? I'd love to help push this forward as well. I've only just stumbled into configuring colors for Ranger, but it seems doable, especially with Dracula as a jumping-off point.

**Reply by dfrico** [reply]
Hi @computergnome99, I made a Catppuccin theme for Ranger [here](https://github.com/dfrico/catppuccin-ranger), but it's not finished and I'm not working on it at the moment.

So far, users have reported [no Frappe flavor](https://github.com/dfrico/catppuccin-ranger/issues/2) and [a possible syntax error](https://github.com/dfrico/catppuccin-ranger/issues/1). The colors weren’t great either: Ranger uses xterm colors, while the Catppuccin palette is hex-based.

I might find some time to keep working on this, but otherwise feel free to contribute to the repo if you're interested.

**Reply by computergnome99** [reply]
Thanks @dfrico. [I created a PR](https://github.com/dfrico/catppuccin-ranger/pull/3) which I think addresses some of these issues. I'm pretty new to Python, so feel free to take a look at my PR and let me know if I should change anything before merging.

**Comment by dfrico** [comment]
We have a [functional ranger theme now](https://github.com/dfrico/catppuccin-ranger). The colors are not perfect because ranger uses a different limited palette (xterm vs hexadecimal) but it's something. I'll leave it open for suggestions and probably submit a PR soon to the official org here.

|  | |
|--------|--------|
| <img width="800" alt="Screenshot 2025-06-30 at 12 02 40" src="https://github.com/user-attachments/assets/5e618817-b43a-46ba-98d6-22fa0b2ce8bd" /> | <img width="800" alt="Screenshot 2025-06-30 at 12 03 11" src="https://github.com/user-attachments/assets/46b70dee-cf32-4c84-80c2-e9c461761530" /> |
| <img width="800" alt="Screenshot 2025-06-30 at 12 06 50" src="https://github.com/user-attachments/assets/7a495712-58cf-4f84-90c6-5e87a8c9f704" /> | <img width="800" alt="Screenshot 2025-06-30 at 12 07 36" src="https://github.com/user-attachments/assets/d2b332f2-56a1-48d9-b312-0f76b0f972c7" /> |
